kvbc/src/categorization/immutable_kv_category.cpp

```diff
@@ -120,14 +131,100 @@ void ImmutableKeyValueCategory::deleteBlock(const ImmutableUpdatesInfo &updates_
   }
 }
 
-std::optional<ImmutableValue> ImmutableKeyValueCategory::get(const std::string &key) const {
-  const auto ser = db_->get(cf_, key);
+std::optional<Value> ImmutableKeyValueCategory::get(const std::string &key, BlockId block_id) const {
+  auto val = getLatest(key);
+  if (!val) {
+    return std::nullopt;
+  }
+  if (asImmutable(val).block_id != block_id) {
```

Comment:
When would this occur? Can this happen outside of a bug or db corruption? Should we at least log it?

Reply:
This is just to conform to a more generic interface that gets the key at a specific block version. However, since immutable category keys can only have one version, if that version is not the requested one, then we report this key is not found.

Reply:
Makes sense. Thanks!